assets/src/hooks/useDetours.ts

```diff
@@ -182,8 +197,10 @@ const subscribeByRoute = (
 
   channel
     .join()
-    .receive("ok", ({ data: data }: { data: SimpleDetour[] }) => {
-      const detoursMap = Object.fromEntries(data.map((v) => [v.id, v]))
+    .receive("ok", ({ data: data }: { data: SimpleDetourData[] }) => {
+      const detoursMap = Object.fromEntries(
+        data.map((v) => [v.id, simpleDetourFromData(v)])
+      )
```

Comment:
issue: since we're needing to rewrite what `useChannel` does so we can support multiple message types ("activated" | "deactivated" etc), this change to use `{ data: data }: { data: SimpleDetourData[] }` is actually hiding an issue that `data` may not actually be a `SimpleDetourData` object. The underlying library has `data` typed as `any`, but it should really be `unknown` which is why we're not getting errors here redefining the type as `SimpleDetourData[]`.

This should instead be using superstruct to validate the `data` before using it.

Reply:
Ohooooooo I see! Thanks for the example, too. Ok I'll need to do this for each of the `.receive("ok",`s and each of the `channel.on`s

Reply:
https://github.com/mbta/skate/pull/2908/commits/69b649c28079d73cc8e1de3e8bb0e84fcabd46a5